pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -853,11 +855,19 @@ type Volume struct {
 	Encrypted *bool `json:"encrypted,omitempty"`
 }
 
+// Cri contains information about the Container Runtimes.
+type Cri struct {
+	// Enable Kubelet support for ContainerD
+	EnableContainerD bool `json:"enableContainerD"`
```

Comment:
I do not like the explicit `enable` in the field/variable name.
Checking some fields of the pod spec how are they defined here https://github.com/kubernetes/kubernetes/blob/fe9073b8c14bbf6bbe42a2259f01d3d3a853a6a3/staging/src/k8s.io/api/core/v1/types.go#L2924-L2947

And I think a simple `containerd` is enough
```suggestion
	Containerd bool `json:"containerd"`
```

Reply:
Instead of using a `bool`, can we rather use something like `Runtime string` that allows either `docker` or `containerd` and defaults to `docker` (even for new clusters for now)? I think API-wise this is cleaner instead of "hard-coding" containerd here.

Reply:
I agree.

Reply:
I Prefer the section:
cri:
    name: containerd
    containerRuntimes:
    - type: gvisor
    - type: kata-containers

Docker is not a CRI (But there are others CRI options like CRI-O. So it is better to make it a string instead of hard-coding boolean but rather leave docker out of it.

Reply:
+1, thanks for pointing it out, agreed

Reply:
I think that `name` would be better in the context of `ContainerRuntime`:
```yaml
cri:
    name: containerd
    containerRuntimes:
    - name: gvisor
    - name: kata-containers
```

Reply:
No strong opinion on this, but in the end we are usually always calling the extension providers with `type`, e.g. `.spec.provider.type`, `spec.extensions[].type`. The only exception is the machine image _name_, but this wasn't done on purpose I guess.

Reply:
Lets keep it to the GEP and the next commit for containerRuntimes.
For now it is:
cri:
  name: containerd